hstream/client/Main.hs

```diff
@@ -0,0 +1,146 @@
+  return $ map simpleCompletion (gs <> cs)
+
+main :: IO ()
+main = do
+  putStrLn "Start Hstream-CLI!"
+  cf <- execParser $ info (parseConfig <**> helper) (fullDesc <> progDesc "start hstream-cli")
+  putStrLn helpInfo
+  runInputT def $ loop cf
+  where
+    loop :: ClientConfig -> InputT IO ()
+    loop c@ClientConfig {..} = handleInterrupt ((liftIO $ putStrLn "Interrupt") >> loop c) $
+      withInterrupt $ do
+        minput <- getInputLine "> "
+        let createRequest api = liftIO $ parseRequest (cHttpUrl ++ ":" ++ show cServerPort ++ api)
+        case minput of
+          Nothing -> return ()
+          Just ":q" -> return ()
+          Just xs -> do
+            case words xs of
+              ":h" : _ -> liftIO $ putStrLn helpInfo
+              "show" : "querys" : _ -> createRequest "/show/querys" >>= handleReq @[TaskInfo] Proxy
+              "delete" : "query" : "all" : _ -> createRequest ("/delete/query/all") >>= handleReq @Resp Proxy
+              "delete" : "query" : dbid -> createRequest ("/delete/query/" ++ unwords dbid) >>= handleReq @Resp Proxy
+              a : sql -> do
+                let val = a : sql
+                case parseAndRefine $ pack $ unwords val of
+                  Left err -> liftIO $ putStrLn $ show err
+                  Right s -> case s of
+                    RQSelect _ -> do
+                      re <- createRequest ("/create/stream/query")
+                      liftIO $
+                        handleStreamReq $
+                          setRequestBodyJSON (ReqSQL (pack $ unwords val)) $
+                            setRequestMethod "POST" re
+                    _ -> do
+                      re <- createRequest "/create/query"
+                      handleReq @(Either String TaskInfo) Proxy $
+                        setRequestBodyJSON (ReqSQL (pack $ unwords val)) $
+                          setRequestMethod "POST" re
+              [] -> return ()
+            loop c
+
+helpInfo :: String
+helpInfo =
+  unlines
+    [ "Command ",
+      "  :h                     help command",
+      "  :q                     quit cli",
+      "  show tasks             list all tasks",
+      "  delete query taskid    delete query by id",
+      "  deletet query all      delete all query",
+      "  sql                    run sql"
+    ]
+
+handleReq :: forall a. (Show a, FromJSON a) => Proxy a -> Request -> InputT IO ()
+handleReq _ req = do
+  (liftIO $ try $ httpBS req) >>= \case
+    Left (e :: SomeException) -> liftIO $ print e
+    Right a -> do
+      case getResponseBody a of
+        "" -> liftIO $ putStrLn "invalid command"
+        ot -> case eitherDecode' (BL.fromStrict ot) of
+          Left e           -> liftIO $ print e
+          Right (rsp :: a) -> liftIO $ pPrint rsp
+
+handleStreamReq :: Request -> IO ()
+handleStreamReq req = do
+  (try $ httpSink req (\_ -> mapM_C BC.putStrLn)) >>= \case
+    Left (e :: SomeException) -> print e
+    Right _                   -> return ()
```

Comment:
Summary:
- Mention your spell and grammar. 
- Make sure that your variable name is clear and understandable.
- There are too many discrete, anonymous functions and unnecessary local variables, which mess up the code. Try to fix them.
- Please try your best to avoid introducing code duplication.